woocommerce/woocommerce
# Add a filter to edit the data-limit value of the term list in the product edit page

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #37548 .

The current attribute term list in the product editor imposes a limitation of 50 terms creating inconvenience for merchants with attributes exceeding this limit. Consequently, they are unable to search for terms from the drop-down menu.



This proposed solution allows for a flexible adjustment of the term limit by introducing a filter, `woocommerce_admin_meta_box_attribute_data_limit`, which will allow to modify the data-limit variable. This approach eliminates the need to edit the plugin files directly, providing a more streamlined and customizable solution.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Import this product with 60 terms: https://d.pr/f/SArtLx
2. If you check the drop-down list in All Product > Product > Attributes, it will be limited to 50
3. To increase the limit to 60 or more, add the following in the functions.php file:

```
function example_callback() {
    return 60;
}
add_filter( 'woocommerce_admin_meta_box_attribute_data_limit', 'example_callback' );
```



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Add a filter to edit the data-limit value of the term list in the product edit page

#### Comment 

</details>
